dev/system-test/firestore.ts

```diff
@@ -2773,19 +2776,21 @@ describe('Bundle building', () => {
 
     const namedQuery = (elements[1] as IBundleElement).namedQuery;
     // Verify saved query.
-    expect(namedQuery).to.deep.equal({
-      name: 'query',
-      readTime: snap.readTime.toProto().timestampValue,
-      // TODO(wuandy): Fix query.toProto to skip undefined fields, so we can stop using `extend` here.
-      bundledQuery: extend(
-        true,
-        {},
-        {
-          parent: query.toProto().parent,
-          structuredQuery: query.toProto().structuredQuery,
-        }
-      ),
-    });
+    expect(namedQuery).to.deep.equal(
+      NamedQuery.fromObject({
+        name: 'query',
+        readTime: snap.readTime.toProto().timestampValue,
+        // TODO(wuandy): Fix query.toProto to skip undefined fields, so we can stop using `extend` here.
+        bundledQuery: extend(
+          true,
+          {},
+          {
+            parent: query.toProto().parent,
+            structuredQuery: query.toProto().structuredQuery,
+          }
+        ),
+      }).toJSON()
```

Comment:
Do we have to do this round-trip here (and below)? Can we not just compare against the resulting JSON directly?

Reply:
The trick here is the expected object we use here will have int64 encoded as number, while the actual object we get from bundles have them encoded as strings, for example.

I have created some utility functions to help with the boilerplate. Unfortunately i cannot figure out how to make them generic, because `fromObject` is static, and returns different types.

Reply:
Is it not possible to manually create the desired Proto output? Would it be possible to write `expect(...).to.deep.equal({integerValue:'1337'}))` instead of `expect(...).to.deep.equal(fromJson({integerValue:1337}).toJSON()))`?

Reply:
This works for the example you give. But the tests deal a lot with timestamps, and `expect(...).to.deep.equal(snap.readTime.toProto.timestampValue)` will compare encoded string `seconds` to the actual integer value. So you get tests failures like:

```
      -    "seconds": "1577836805"
      +    "seconds": 1577836805
```